plugins/colorsource/export.js

```diff
@@ -0,0 +1,252 @@
+        manufacturerName: fixture.manufacturer.name,
+        maxOffset: mode.channels.length - 1,
+        modeName: mode.name,
+        modelName: fixture.name,
+        parameters: []
+      };
+
+      mode.channels.forEach((channel, channelIndex) => {
+        const name = channel.name;
+
+        if (channel instanceof SwitchingChannel) {
+          channel = channel.defaultChannel;
+        }
+
+        const channelJson = {
+          coarse: channelIndex,
+          fadeWithIntensity: false,
+          fine: null,
+          highlight: 65535,
+          home: 0,
+          invert: false,
+          name,
+          ranges: [],
+          size: 8,
+          snap: false,
+          type: getCSChannelType(channel)
+        };
+
+        if (channelJson.type === 5) {
+          channelJson.name = channelJson.name.replace(/ /g, ``); // e.g. 'Warm White' -> 'WarmWhite'
+        }
+
+        if (channel instanceof FineChannel) {
+          if (channel.resolution === CoarseChannel.RESOLUTION_16BIT) {
+            // already handled by "fine" attribute of coarse channel
+            return;
+          }
+
+          channelJson.type = 3;
+          channelJson.home = scaleDmxValue(channel.defaultValue, CoarseChannel.RESOLUTION_8BIT, CoarseChannel.RESOLUTION_16BIT);
+        }
+        else {
+          addChannelDetails(channelJson, channel, channelIndex);
+        }
+
+        // remove null values and empty arrays
+        Object.entries(channelJson).forEach(([key, value]) => {
+          if (value === null || (Array.isArray(value) && value.length === 0)) {
+            delete channelJson[key];
+          }
+        });
+
+        fixtureJson.parameters.push(channelJson);
+      });
+
+      fixtureJson.colortable = getColorTable(fixtureJson.parameters);
+
+      // remove null values and empty arrays
+      Object.entries(fixtureJson).forEach(([key, value]) => {
+        if (value === null || (Array.isArray(value) && value.length === 0)) {
+          delete fixtureJson[key];
+        }
+      });
+
+      exportJson.personalities.push(fixtureJson);
+
+
+      /**
+       * @param {!AbstractChannel} channel The OFL channel of which the ColorSource channel type should be returned.
+       * @returns {number} One of ColorSource's channel types as positive integer.
+       */
+      function getCSChannelType(channel) {
+        if (channel.type === `NoFunction`) {
+          return 0;
+        }
+
+        if (channel.type === `Single Color` || [`Hue`, `Saturation`].includes(channel.name)) {
+          return 5;
+        }
+
+        if (channel.type === `Intensity`) {
+          return 1;
+        }
+
+        if ([`Pan`, `Tilt`].includes(channel.type)) {
+          return 2;
+        }
+
+        if (channel.type === `Color Temperature`) {
+          return 3;
+        }
+
+        return 4;
+      }
+
+      /**
+       * Adds detailed information to given channel JSON that only a CoarseChannel can provide.
```

Comment:
Please mention that it also fills `fixtureJson.commands`.

Reply:
Moved command generation to another function.